libcloud/storage/base.py

```diff
@@ -383,28 +408,32 @@ def download_object(self, obj, destination_path, overwrite_existing=False,
             'download_object not implemented for this driver')
 
     def download_object_as_stream(self, obj, chunk_size=None):
+        # type: (Object, Optional[int]) -> Iterator[bytes]
         """
-        Return a generator which yields object data.
+        Return a iterator which yields object data.
 
         :param obj: Object instance
-        :type obj: :class:`Object`
+        :type obj: :class:`libcloud.storage.base.Object`
 
         :param chunk_size: Optional chunk size (in bytes).
         :type chunk_size: ``int``
+
+        :rtype: ``iterator`` of ``bytes``
         """
         raise NotImplementedError(
             'download_object_as_stream not implemented for this driver')
 
     def upload_object(self, file_path, container, object_name, extra=None,
                       verify_hash=True, headers=None):
+        # type: (str, Container, str, Optional[dict], Optional[bool], Optional[dict]) -> Object  # noqa: E501
```

Comment:
Since ``verify_hash`` has a default value, I don't think it needs ``Optional`` annotation.

Reply:
And for ``headers`` we could probably use ``Dict[str, str]``.

Reply:
Done in d80b17c.